### PR TITLE
Added IDA* implementation

### PR DESCRIPTION
**Pull request by tsa87**
Implemented Iterative Deepening A* Search in `search.py`

### CODE REVIEW

**Comment by tsa87** [comment]
First open-source contribution :)
Please advise revisions.

File: search.py [hunk 1]
```diff
@@ -420,8 +420,35 @@ def astar_search(problem, h=None, display=False):
     return best_first_graph_search(problem, lambda n: n.path_cost + h(n), display)
 
 
+def cost_limited_astar_search(problem, limit, f):
```
Comment (antmarakis) [inline search.py]: You need to comment the code a bit, unfortunately right now it may be a bit hard to follow.

File: search.py [hunk 2]
```diff
@@ -420,8 +420,35 @@ def astar_search(problem, h=None, display=False):
...
+
+    def recursive_clas(node, problem, limit, f):
+        if problem.goal_test(node.state):
```
Comment (antmarakis) [inline search.py]: For example, add some description for each if statement

File: search.py [hunk 3]
```diff
@@ -420,8 +420,35 @@ def astar_search(problem, h=None, display=False):
     return best_first_graph_search(problem, lambda n: n.path_cost + h(n), display)
 
 
+def cost_limited_astar_search(problem, limit, f):
+
+    def recursive_clas(node, problem, limit, f):
```
Comment (antmarakis) [inline search.py]: Is that a type? What is 'clas'?

File: search.py [hunk 4]
```diff
@@ -420,8 +420,35 @@ def astar_search(problem, h=None, display=False):
...
+        if problem.goal_test(node.state):
+            return node
+        elif f(node) > limit:
+            return 'cutoff'
+        else:
+            cutoff_occurred = False
+            for child in node.expand(problem):
```
Comment (antmarakis) [inline search.py]: Here it would also be nice to have a quick comment explaining what is going on

File: search.py (hunk 1 above)
Comment (tsa87) [reply search.py]: Hey, thanks for the feedback! I added more comments and descriptions for the search function. Please feel free to let me know if they are not clear.

File: search.py (hunk 3 above)
Comment (tsa87) [reply search.py]: I changed the function name to "recursive_cost_limited_astar_search"
